inspirehep/modules/migrator/models.py

```diff
@@ -38,6 +38,10 @@
 class LegacyRecordsMirror(db.Model):
     __tablename__ = 'legacy_records_mirror'
 
+    __table_args__ = (
+        db.Index('ix_legacy_records_mirror_valid_collection', 'valid', 'collection'),
+    )
+
     recid = db.Column(db.Integer, primary_key=True, index=True)
     last_updated = db.Column(db.DateTime, default=datetime.utcnow, nullable=False, index=True)
     _marcxml = db.Column('marcxml', db.LargeBinary, nullable=False)
```

Comment:
as I was saying, the whole point of this is that you can now drop the index in `valid` in the next line, because it performs part of the same function as the new index you're adding. You also need to add a migration to remove/add that index on upgrade/downgrade.